searx/engines/wikicommons.py

```diff
@@ -1,102 +1,206 @@
+        "gsrsearch": f"filetype:{filetype} {query}",
+        # imageinfo: https://commons.wikimedia.org/w/api.php?action=help&modules=query%2Bimageinfo
+        "iiprop": "url|size|mime",
+        "iiurlheight": "180",  # needed for the thumb url
     }
+    params["url"] = f"{wc_api_url}?{urlencode(args, safe=':|')}"
 
-    params["url"] = f"{base_url}/w/api.php{search_prefix}&{urlencode(args, safe=':|')}"
-    return params
 
+def response(resp: "SXNG_Response") -> EngineResults:
 
-def response(resp):
-    results = []
-    json = resp.json()
+    res = EngineResults()
+    json_data = resp.json()
+    pages = json_data.get("query", {}).get("pages", {}).values()
 
-    if not json.get("query", {}).get("pages"):
-        return results
-    for item in json["query"]["pages"].values():
+    for item in pages:
+
+        if not item.get("imageinfo", []):
+            continue
         imageinfo = item["imageinfo"][0]
-        title = item["title"].replace("File:", "").rsplit('.', 1)[0]
-        result = {
-            'url': imageinfo["descriptionurl"],
-            'title': title,
-            'content': html_to_text(item["snippet"]),
-        }
-
-        if search_type == "images":
-            result['template'] = 'images.html'
-            result['img_src'] = imageinfo["url"]
-            result['thumbnail_src'] = imageinfo["thumburl"]
-            result['resolution'] = f'{imageinfo["width"]} x {imageinfo["height"]}'
-        else:
-            result['thumbnail'] = imageinfo["thumburl"]
-
-        if search_type == "videos":
-            result['template'] = 'videos.html'
-            if imageinfo.get('duration'):
-                result['length'] = datetime.timedelta(seconds=int(imageinfo['duration']))
-            result['iframe_src'] = imageinfo['url']
-        elif search_type == "files":
-            result['template'] = 'files.html'
-            result['metadata'] = imageinfo['mime']
-            result['size'] = humanize_bytes(imageinfo['size'])
-        elif search_type == "audio":
-            result['iframe_src'] = imageinfo['url']
-
-        results.append(result)
-
-    return results
+
+        title: str = item["title"].replace("File:", "").rsplit(".", 1)[0]
+        content = html_to_text(item["snippet"])
+
+        url: str = imageinfo["descriptionurl"]
+        media_url: str = imageinfo["url"]
+        mimetype: str = imageinfo["mime"]
+        thumbnail: str = imageinfo["thumburl"]
+        size = imageinfo.get("size")
+        if size:
+            size = humanize_bytes(size)
+
+        duration = None
+        seconds: str = imageinfo.get("duration")
+        if seconds:
+            try:
+                duration = datetime.timedelta(seconds=int(seconds))
+            except OverflowError:
+                pass
+
+        if wc_search_type == "file":
+            res.add(
+                res.types.File(
+                    title=title,
+                    url=url,
+                    content=content,
```

Comment:
```suggestion
                    content=content,
                    filename=title,
```
I don't know if that makes sense because it'd be displayed twice then, but perhaps it's worth setting the `filename` too.

Reply:
The `title` is not suitable because the file extension has already been removed.  The filename in `media_url` is more suitable --> [diff](https://github.com/searxng/searxng/compare/46e96496c2ef7a83c77e2f0453b1e66aecc85e58..f57f91f31e8c90baefc9831c5504ae3763d8a9b6)